internal/adapter/sqlite/note_dao.go

```diff
@@ -421,9 +420,10 @@ func (d *NoteDAO) expandMentionsIntoMatch(opts core.NoteFindOpts) (core.NoteFind
 	}
 
 	// Expand the mention queries in the match predicate.
-	match := opts.Match.String()
-	match += " " + strings.Join(mentionQueries, " OR ")
-	opts.Match = opt.NewString(match)
+	match := make([]string, len(opts.Match))
+	copy(match, opts.Match)
+	match = append(match, " ("+strings.Join(mentionQueries, " OR ")+") ")
+	opts.Match = match
```

Comment:
Not codebase-fluent (or go-fluent) enough to know whether we should copy here to avoid altering `x.Match` in other terms. Sort of just wrote it with a mind to check later.

Reply:
Actually don't think it's need as append does its own copy.